library/src/main/java/org/apache/apex/malhar/lib/state/managed/IncrementalCheckpointManager.java

```diff
@@ -188,8 +190,12 @@ protected void committed(int operatorId, long windowId) throws IOException, Inte
   {
     LOG.debug("data manager committed {}", windowId);
     for (Long currentWindow : savedWindows.keySet()) {
+      if (currentWindow <= smallestPendingWindow) {
+        continue;
+      }
       if (currentWindow <= windowId) {
-        LOG.debug("to transfer {}", windowId);
+        LOG.debug("to transfer {}", currentWindow);
+        smallestPendingWindow = currentWindow;
```

Comment:
can the name be changed of smallestPendingWindow. It can be slightly misleading.